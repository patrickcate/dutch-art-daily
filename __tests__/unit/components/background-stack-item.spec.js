import { shallowMount } from '@vue/test-utils'
import mockData from '@fixtures/mock-data.js'
import BackgroundStackItem from '@components/background-stack-item.vue'

describe('BackgroundStackItem Component', () => {
  it('is a Vue instance', () => {
    const wrapper = shallowMount(BackgroundStackItem, {
      propsData: {
        id: '01-01',
      },
      mocks: {
        $store: {
          state: {
            activeId: '01-01',
          },
          getters: {
            getArtworkById() {
              return mockData.artwork
            },
          },
        },
      },
    })

    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    const wrapper = shallowMount(BackgroundStackItem, {
      propsData: {
        id: '01-01',
      },
      mocks: {
        $store: {
          state: {
            activeId: '01-01',
          },
          getters: {
            getArtworkById() {
              return mockData.artwork
            },
          },
        },
      },
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('has no active background', () => {
    const wrapper = shallowMount(BackgroundStackItem, {
      propsData: {
        id: '01-01',
      },
      mocks: {
        $store: {
          state: {
            activeId: '',
          },
          getters: {
            getArtworkById() {
              return null
            },
          },
        },
      },
    })

    expect(wrapper).toMatchSnapshot()
  })
})
