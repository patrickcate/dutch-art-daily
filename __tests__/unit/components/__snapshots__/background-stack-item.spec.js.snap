// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundStackItem Component has no active background 1`] = `<div class="background-stack__item"></div>`;

exports[`BackgroundStackItem Component renders correctly 1`] = `<div class="background-stack__item background-stack__item--today" style="background-image: url(/photos/01-01/01-01--lqi-d0wAmzwM620.jpg);"></div>`;
